clj-refactor.el

```diff
@@ -91,13 +91,17 @@ paths once this flag is removed."
   :safe #'booleanp)
 
 (defcustom cljr-magic-require-namespaces
-  '(("io"   . "clojure.java.io")
+  '(("edn"  . "clojure.edn")
+    ("io"     "clojure.java.io" :only ("clj"))
     ("math" . "clojure.math")
     ("set"  . "clojure.set")
     ("str"  . "clojure.string")
     ("walk" . "clojure.walk")
     ("zip"  . "clojure.zip"))
-  "Alist of aliases and namespaces used by `cljr-slash'."
+  "Alist of aliases to namespace libspec recommendations for `\\[cljr-slash]'.
+
+An optional keyword `:only` can limit a recommendation to the set of
+language contexts the libspec is available in."
```

Comment:
Ideally, list all possible values of :only here

Reply:
That's up to the middleware, so once it's documented there we can link to accepted values. We can give clj/cljs as examples, but I don't think it makes sense to document the complete list from the client side.